backend/plan/models.py

```diff
@@ -43,11 +43,23 @@ class Schedule(models.Model):
         """
         ),
     )
+
+    advanced_registration = models.BooleanField(
+        blank=False,
+        default=False,
+        help_text=dedent(
+            """
+            A label to denote whether the user's schedule was made for advanced registration.
+            """
+        ),
+    )
+
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
 
     class Meta:
         unique_together = (("name", "semester", "person"),)
+        unique_together = (("name", "semester", "person", "advanced_registration"), ("semester", "person", "advanced_registration"))
```

Comment:
Each tuple is `and`ed together -- so both tuples have to be unique, otherwise everything fails. The first tuple ends up being redundant because it just a subset of the cases covered by the second tuple.

I would fix this by deleting the second constraint. When advanced registration is False (all of the cases prior to this PR, we have the original constraint of `("name", "semester", "person"))`. When `advanced_registration` is true, we should only have 1 schedule per semester so the `("name", "semester", "person")` constraint should always succeed.